### PR TITLE
Search api

### PR DESCRIPTION
**Pull request by kiverso**
Adds ability to search the api for a podcast. A user can then submit the podcast to podsmack from these results. Adds tests and makes podcast_uri a unique field.

### CODE REVIEW

File: app/javascript/packs/application.js [hunk 1]
```diff
@@ -18,3 +18,5 @@ require("channels")
 //
 // const images = require.context('../images', true)
 // const imagePath = (name) => images(name, true)
+import "bootstrap"
+import "../stylesheets/application"
```
Comment (takeller) [inline app/javascript/packs/application.js:22]: Should this be "./stylesheets/application"

File: app/controllers/podcasts_controller.rb [hunk 2]
```diff
@@ -22,22 +22,20 @@ def show
...
+    podcast = current_user.podcasts.new(podcast_params)
     if podcast.save
-      params[:podcast][:tags].each {|tag| PodcastTag.create({podcast_id: podcast.id, tag_id: tag})}
+      params[:tags].each {|tag| PodcastTag.create({podcast_id: podcast.id, tag_id: tag})} if params[:tags]
       flash[:notice] = 'Podcast submitted and waiting approval'
       redirect_to '/dashboard'
     else
       flash[:error] = podcast.errors.full_messages.to_sentence
-      redirect_to new_user_podcast_path(user)
+      redirect_to request.referrer
```
Comment (janegreene) [inline app/controllers/podcasts_controller.rb:32]: This is so handy, nice trick.

File: app/models/podcast.rb [hunk 3]
```diff
@@ -5,6 +5,8 @@ class Podcast < ApplicationRecord
   validates_presence_of :name
   validates_presence_of :location
   validates_presence_of :description
+  validates_presence_of :podcast_uri
+  validates_uniqueness_of :podcast_uri
```
Comment (janegreene) [inline app/models/podcast.rb:9]: This should prevent duplicates from importing.

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -15,7 +15,7 @@
 
 <h4> Podcast Application </h4>
 
-<%= button_to 'Upload your own podcast', new_user_podcast_path(@user), method: 'get', class: "btn btn-outline btn-xl js-scroll-trigger" %>
+<%= button_to 'Upload your own podcast', search_path, method: 'get', class: "btn btn-outline btn-xl js-scroll-trigger" %>
```
Comment (janegreene) [inline app/views/users/show.html.erb:18]: Awesome feature, nice job!

File: app/javascript/packs/application.js (hunk 1 above)
Comment (muydanny) [reply app/javascript/packs/application.js]: EAGLE EYE

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -72,3 +72,9 @@
   # arbitrary gems may also be filtered via:
   # config.filter_gems_from_backtrace("gem name")
 end
+
+VCR.configure do |config|
```
Comment (janegreene) [inline spec/rails_helper.rb:76]: Thanks for showing us this in action.

**Pull request merged**